service/resource/configmap/current.go

```diff
@@ -18,27 +17,25 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for config maps in the Kubernetes API")
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for a list of config maps in the Kubernetes API")
 
-	var configMaps []*apiv1.ConfigMap
-
-	namespace := key.ClusterNamespace(customObject)
-	configMapNames := key.ConfigMapNames(customObject)
-
-	for _, name := range configMapNames {
-		manifest, err := r.k8sClient.CoreV1().ConfigMaps(namespace).Get(name, apismetav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "did not find a config map in the Kubernetes API")
-			// fall through
-		} else if err != nil {
+	var currentConfigMaps []*apiv1.ConfigMap
+	{
+		namespace := key.ClusterNamespace(customObject)
+		configMapList, err := r.k8sClient.CoreV1().ConfigMaps(namespace).List(apismetav1.ListOptions{})
+		if err != nil {
 			return nil, microerror.Mask(err)
 		} else {
-			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "found a config map in the Kubernetes API")
-			configMaps = append(configMaps, manifest)
+			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "found a list of config maps in the Kubernetes API")
+
+			for _, item := range configMapList.Items {
+				c := item
+				currentConfigMaps = append(currentConfigMaps, &c)
```

Comment:
This pointer trick has to be done because the list items are no pointer but we want to have pointers and taking the pointer directly from the loop results in having always the same resources.